l5kit/l5kit/cle/metrics.py

```diff
@@ -23,6 +25,84 @@ def compute(self, simulation_output: SimulationOutput) -> torch.Tensor:
         raise NotImplementedError
 
 
+class CollisionMetricBase(ABC):
+    """This is the abstract base class for the collision metric.
+
+    :param collision_type: the type of collision to compute
+    """
+    @abstractmethod
+    def __init__(self, collision_type: l5metrics.CollisionType) -> None:
+        self.collision_type = collision_type
+
+    def _compute_frame(self, recorded_agents: np.ndarray,
+                       simulated_frame_ego_state: torch.Tensor) -> float:
+        """Detects collision per frame of the scene.
+
+        :param observed_frame: the ground-truth frame
+        :param simulated_frame_ego_state: ego state from the simulated frame,
+                                          this is a 1D array with the frame
+                                          ego state
+        :returns: metric result for the frame, where 1 means a collision,
+                  and 0 otherwise.
+        """
+        simulated_centroid = simulated_frame_ego_state[:TrajectoryStateIndices.Y + 1].cpu().numpy()
+        simulated_angle = simulated_frame_ego_state[TrajectoryStateIndices.THETA].cpu().numpy()
+        simulated_extent = np.r_[EGO_EXTENT_LENGTH, EGO_EXTENT_WIDTH]
```

Comment:
this is the same as `np.asarray([EGO_EXTENT_LENGTH, EGO_EXTENT_WIDTH])` right?

Reply:
Yes, it is just a short hand to look closer to how lists are built in Python.